include/pressio/ops/kokkos/ops_fill.hpp

```diff
@@ -53,12 +53,20 @@
 
 namespace pressio{ namespace ops{
 
-template <typename T>
+template <typename T, typename ScalarType>
+#ifdef PRESSIO_ENABLE_CXX20
+requires (::pressio::is_native_container_kokkos<T>::value
+       or ::pressio::is_expression_acting_on_kokkos<T>::value)
+       and std::convertible_to<ScalarType, typename ::pressio::Traits<T>::scalar_type>
+void
+#else
 ::pressio::mpl::enable_if_t<
     (::pressio::is_native_container_kokkos<T>::value
   or ::pressio::is_expression_acting_on_kokkos<T>::value)
+  and std::is_convertible<ScalarType, typename ::pressio::Traits<T>::scalar_type>::value
   >
-fill(const T & v, typename ::pressio::Traits<T>::scalar_type value)
+#endif
+fill(const T & v, const ScalarType & value)
 {
   ::KokkosBlas::fill(impl::get_native(v), value);
```

Comment:
why are you not doing the explicit construction of `v` here like you did for the other types?

Reply:
Thanks for catching and I'm sorry for the omission... Fixed on e556ee75a577e25a42a30c3f4d05447662bc3faf.